packages/react/src/components/Flex/FlexItem.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+import cx from 'classnames'
+import { UIComponent, commonPropTypes } from '../../lib'
+import { mergeStyles } from '../../lib/mergeThemes'
+import { Extendable } from '../../types'
+
+export type FlexItemSize = 'size.half' | 'size.quarter' | 'size.small' | 'size.medium'
+
+export interface FlexItemProps {
+  /** Controls item's alignment. */
+  align?: 'auto' | 'start' | 'end' | 'center' | 'baseline' | 'stretch'
+
+  /** Defines size of the item. */
+  size?: FlexItemSize
+
+  /**
+   * Item can fill remaining space of the container.
+   * If numeric value is provided, remaining space will be distributed proportionally between all the items.
+   * */
+  grow?: boolean | number
+
+  /**
+   * Controls item's ability to shrink.
+   * */
+  shrink?: boolean | number
+
+  /**
+   * Item can be pushed towards opposite side in the container's direction.
+   */
+  push?: boolean
+
+  /**
+   * IGNORE (will be refactored and not exposed via API).
+   * Value is automatically set by parent Flex component.
+   */
+  flexDirection?: 'row' | 'column'
+}
+
+class FlexItem extends UIComponent<Extendable<FlexItemProps>> {
+  static className = 'ui-flex__item'
+
+  static displayName = 'FlexItem'
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      content: false,
+    }),
+    align: PropTypes.oneOf(['auto', 'start', 'end', 'center', 'baseline', 'stretch']),
+    size: PropTypes.string,
```

Comment:
Why not more strict prop type here?

Reply:
dev reasons - would like to omit duplication before final type was settled, Thanks!